driver-dse/src/main/java/com/datastax/driver/auth/DseAuthProvider.java

```diff
@@ -65,20 +91,153 @@
  *       principal="user@MYDOMAIN.COM";
  * };
  * </pre>
- * <p/>
- * To connect to clusters using internal authentication, use the standard method for
- * setting credentials. eg:
- * <p/>
- * <pre>
- * Cluster cluster = Cluster.builder()
- *                          .addContactPoint(hostname)
- *                          .withCredentials("username", "password")
- *                          .build();
- * </pre>
  */
 public class DseAuthProvider implements AuthProvider {
-    @Override
+    private static final String DSE_AUTHENTICATOR = "com.datastax.bdp.cassandra.auth.DseAuthenticator";
+    private final String username;
+    private final String password;
+
+    public DseAuthProvider() {
+        this.username = null;
+        this.password = null;
+    }
+
+    public DseAuthProvider(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
+
     public Authenticator newAuthenticator(InetSocketAddress host, String authenticator) throws AuthenticationException {
-        return new KerberosAuthenticator(host);
+        // Need both username and password for PLAIN auth
+        if ((username != null) && (password != null)) {
+            return new PlainTextAuthenticator(authenticator, username, password);
+        }
+        return new GSSAPIAuthenticator(authenticator, host);
+    }
+
+    public static abstract class BaseDseAuthenticator implements Authenticator {
```

Comment:
This was left public so that we can extend the ```DseAuthProvider``` and provide additional ```Authenticator``` implementations